src/core/webidl.js

```diff
@@ -5,181 +5,112 @@
-function makeMarkup(parse, { suppressWarnings } = {}) {
-  const templates = {
-    wrap(items) {
-      return items
-        .reduce(flatten, [])
-        .map(x => (typeof x === "string" ? new Text(x) : x));
-    },
-    trivia(t) {
-      if (!t.trim()) {
-        return t;
-      }
-      return hyperHTML`<span class='idlSectionComment'>${t}</span>`;
-    },
-    generic(wrapped) {
-      if (standardTypes.has(wrapped)) {
-        return hyperHTML`<a data-cite='${standardTypes.get(
-          wrapped
-        )}'>${wrapped}</a>`;
-      }
-      return hyperHTML`<a data-link-for="">${wrapped}</a>`;
-    },
-    reference(wrapped, name) {
-      if (standardTypes.has(name)) {
-        return hyperHTML`<a data-cite='${standardTypes.get(
-          name
-        )}'>${wrapped}</a>`;
-      }
-      return hyperHTML`<a data-link-for="">${wrapped}</a>`;
-    },
-    name(escaped, { data, parent }) {
-      if (data.idlType && data.idlType.type === "argument-type") {
-        return hyperHTML`<span class="idlParamName">${escaped}</span>`;
-      }
-      const parentName = parent ? parent.name : "";
-      const { name } = getNameAndId(data, parentName);
-      const dfn = findDfn(data, name, {
-        parent: parentName,
-        suppressWarnings,
-      });
-      const idlAnchor = createIdlAnchor(escaped, data, parentName, dfn);
-      const className = parent ? "idlName" : "idlID";
-      if (data.type === "enum-value") {
-        return idlAnchor;
-      }
-      return hyperHTML`<span class="${className}">${idlAnchor}</span>`;
-    },
-    type: contents => hyperHTML`<span class="idlType">${contents}</span>`,
-    inheritance: contents =>
-      hyperHTML`<span class="idlSuperclass">${contents}</span>`,
-    definition: createIdlElement,
-    extendedAttribute: contents =>
-      hyperHTML`<span class="extAttr">${contents}</span>`,
-    extendedAttributeReference(name) {
-      if (!extendedAttributesLinks.has(name)) {
-        return hyperHTML`<a>${name}</a>`;
-      }
-      return hyperHTML`<a data-cite="${extendedAttributesLinks.get(
-        name
-      )}">${name}</a>`;
-    },
-  };
-  const result = webidl2.write(parse, { templates });
-  return hyperHTML`<pre class="def idl">${result}</pre>`;
-}
+const templates = {
+  wrap(items) {
+    return items
+      .reduce(flatten, [])
+      .filter(x => x !== "")
+      .map(x => (typeof x === "string" ? new Text(x) : x));
+  },
+  trivia(t) {
+    if (!t.trim()) {
+      return t;
+    }
+    return hyperHTML`<span class='idlSectionComment'>${t}</span>`;
+  },
+  generic(keyword) {
+    // Interfaces start with capital letters, like Promise, FrozenArray
+    return /^[A-Z]/.test(keyword)
+      ? hyperHTML`<a data-xref-type="_IDL_">${keyword}</a>`
+      : // Other keywords like sequence, maplike, etc...
+        hyperHTML`<a data-xref-type="dfn">${keyword}</a>`;
+  },
+  reference(wrapped) {
+    return hyperHTML`<a data-xref-type="_IDL_">${wrapped}</a>`;
```

Comment:
Just curious, does shepherd assumes primitive types as `_IDL_`, as `reference()` should also cover them?

Reply:
@sidvishnoi can confirm, but I think: "_CONCEPT_" and "_IDL_" are an abstraction we created. 

Shepherd classifies:
attribute, dfn, dict-member, dictionary, element, enum, enum-value, event, exception, extended-attribute, interface, method, typedef

Reply:
@saschanaz, to be more clear, if `reference()` was passed the actual types above, then  that's potentially a nice optimization for xref... as it knows exactly what to look for. @sidvishnoi, wdty?

Reply:
We send `_IDL_` when we are not sure of exactly what IDL type it is. If we're sure it's a dict-memmber, enum etc, we should use that instead. `_IDL_` is for fallback only. The types are defined [sidvishnoi/respec-xref-route](https://github.com/sidvishnoi/respec-xref-route/blob/5538b9e022f9073e1a55124f01a25489ecb02176/utils.js#L48).

Reply:
> if reference() was passed the actual types above, then that's potentially a nice optimization for xref... as it knows exactly what to look for.

Primitive types are primitive, they do not belong to any of those groups in [sidvishnoi/respec-xref-route](https://github.com/sidvishnoi/respec-xref-route/blob/5538b9e022f9073e1a55124f01a25489ecb02176/utils.js#L48), so for now, no.